server/routes/change-of-circumstances/handlers/changeOfCircumstanceRoutes.ts

```diff
@@ -0,0 +1,53 @@
+import { Request, Response } from 'express'
+import { getPagination, PaginationRequest } from '../../../utils/paginationUtils'
+import ActivitiesService from '../../../services/activitiesService'
+import PrisonService from '../../../services/prisonService'
+import logger from '../../../../logger'
+
+const ITEMS_PER_PAGE = 10
+
+export default class ChangeOfCircumstanceRoutes {
+  constructor(private readonly activitiesService: ActivitiesService, private readonly prisonService: PrisonService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const { page, date } = req.query
+    const prison = user.activeCaseLoadId
+    const queryDate = date as string
+
+    // Get a page of results and call search for additional details
+    const pageNo = +page || 0 // Default to the first page if not set
+    const searchResults = await this.activitiesService.getChangeEvents(prison, queryDate, pageNo, ITEMS_PER_PAGE, user)
+    const prisonerNumbers = searchResults.content.map(e => e.prisonerNumber)
+    const prisoners = await this.prisonService.searchInmatesByPrisonerNumbers(prisonerNumbers, user)
+
+    // Get the pagination settings for this list
+    const totalResults = searchResults.totalElements || 5 // TODO: Temporary until API returns this!!
+    const currentPage = searchResults.pageNumber
```

Comment:
This TODO will be addressed when the API change is merged - then the default of 5 can be removed. The API is not currently returning the total number of items found.

Reply:
API amended to return the totalElements - so another PR to remove the 5 default coming now.